File: bin/sp-4.4.1/sicstus-4.4.1/se/sics/prologbeans/PBAtomic.java
/**
 * Copyright (c) 2003 SICS AB. All rights reserved.
 */
package se.sics.prologbeans;

import java.io.IOException;

/**
 * <code>PBAtomic</code> is the representation
 * of Prolog constants.
 */

abstract class PBAtomic extends PBTerm {

	final static int ATOM = 1;
	final static int INTEGER = 2;
	final static int FLOAT = 3;
	final static int VARIABLE = 4;

	// private final int type;

	/**
	 * Creates a new <code>PBAtomic</code> instance with the specified name.
	 */
	PBAtomic(String name) {
		super(name);
	}

	abstract int getType();

	public boolean isAtom() {
		return false;
	}

	public boolean isInteger() {
		return false;
	}

	public boolean isFloat() {
		return false;
	}

	public boolean isAtomic() {
		return true;
	}

	abstract String toPrologString();

	abstract public String toString();

	void fastWrite(FastWriter writer) throws IOException {
		writer.writeAtomic(this);
	}

	// @Override
	PBTerm[] fastWritePrefix(FastWriter writer) throws IOException {
		writer.writeAtomic(this);
		return PBTerm.NO_TERMS;
	}

}

File: bin/sp-4.4.1/sicstus-4.4.1/library/prologbeans.NET/PBAtomic.cs
// Copyright (c) 2004 SICS AB. All rights reserved.
// 
namespace se.sics.prologbeans
{
	/// <summary> <code>PBAtomic</code> is the representation
	/// of Prolog constants.
	/// </summary>
	
	abstract class PBAtomic:PBTerm
	{
		internal abstract int Type{ get; }

		override sealed public bool Atomic {
			get {
				return true;
			}
		}

		internal const int ATOM = 1;
		internal const int INTEGER = 2;
		internal const int FLOAT = 3;
		internal const int VARIABLE = 4;

		/// <summary> Creates a new <code>PBAtomic</code> instance with the specified name.</summary>
		internal PBAtomic (string name) : base (name)
		{
		}

		internal abstract override string toPrologString ();

		public abstract override string ToString ();

		internal override PBTerm[] fastWritePrefix (FastWriter writer)
		{
			writer.writeAtomic (this);
			return PBTerm.NO_TERMS;
		}
	}
}
